### PR TITLE
bpo-46725: Document starred expressions in for statements

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue46725

Automerge-Triggered-By: GH:pablogsal

### CODE REVIEW

File: Doc/reference/compound_stmts.rst [hunk 1]
```diff
@@ -154,17 +154,20 @@ The :keyword:`for` statement is used to iterate over the elements of a sequence
...
+iterator is created for the result of the ``starred_list``.  The expression
+list can contain starred elements (``*x, *y``) that will be unpacked in the
+final iterator (as when constructing a ``tuple`` or ``list`` literal). The
+suite is then executed once for each item provided by the iterator, in the
+order returned by the iterator.  Each item in turn is assigned to the target
+list using the standard rules for assignments (see :ref:`assignment`), and then
+the suite is executed.  When the items are exhausted (which is immediately when
+the sequence is empty or an iterator raises a :exc:`StopIteration` exception),
+the suite in the :keyword:`!else` clause, if present, is executed, and the loop
+terminates.
```
Comment (JelleZijlstra) [inline Doc/reference/compound_stmts.rst:170]: Should also add a `versionchanged` clause.

**Review by JelleZijlstra — COMMENTED** [review]
Should we also add unit tests that test this behavior?

File: Lib/test/test_grammar.py [hunk 2]
```diff
@@ -1394,6 +1394,12 @@ def __getitem__(self, i):
             result.append(x)
         self.assertEqual(result, [1, 2, 3])
 
+        result = []
+        a = b = c = [1,2,3]
```
Comment (gvanrossum) [inline Lib/test/test_grammar.py]: PEP 8 please. :-)
```suggestion
        a = b = c = [1, 2, 3]
```

File: Doc/whatsnew/3.11.rst [hunk 3]
```diff
@@ -160,6 +160,9 @@ traceback. (Contributed by Irit Katriel in :issue:`45607`.)
 Other Language Changes
 ======================
 
+* Starred expressions can be used in :ref:`for statements<for>` (See
+  :issue:`46725` for more details).
```
Comment (gvanrossum) [inline Doc/whatsnew/3.11.rst]: Punctuation nit?

```suggestion
* Starred expressions can be used in :ref:`for statements<for>`. (See
  :issue:`46725` for more details.)
```

**Review by gvanrossum — APPROVED** [review]

**Pull request merged**